src/main/java/uk/gov/hmcts/dts/fact/services/search/PostcodeSearchForServiceAreaRunner.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.hmcts.dts.fact.services.search;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.dts.fact.entity.CourtWithDistance;
+import uk.gov.hmcts.dts.fact.entity.ServiceArea;
+import uk.gov.hmcts.dts.fact.mapit.MapitData;
+import uk.gov.hmcts.dts.fact.model.CourtReferenceWithDistance;
+import uk.gov.hmcts.dts.fact.repositories.CourtWithDistanceRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+import static uk.gov.hmcts.dts.fact.model.ServiceAreaType.CIVIL;
+import static uk.gov.hmcts.dts.fact.model.ServiceAreaType.FAMILY;
+
+@Component
+public class PostcodeSearchForServiceAreaRunner {
+
+    private static final String LOCAL_AUTHORITY = "local-authority";
+    private static final String REGIONAL = "regional";
+
+    private final CourtWithDistanceRepository courtWithDistanceRepository;
+
+    public PostcodeSearchForServiceAreaRunner(CourtWithDistanceRepository courtWithDistanceRepository) {
+        this.courtWithDistanceRepository = courtWithDistanceRepository;
+    }
+
+    public List<CourtReferenceWithDistance> getSearchFor(final ServiceArea serviceArea,
+                                                         final String postcode,
+                                                         final MapitData mapitData) {
+
+        final String type = serviceArea.getType();
+        final String areaOfLaw = serviceArea.getAreaOfLaw().getName();
+        final Double lat = mapitData.getLat();
+        final Double lon = mapitData.getLon();
+
+        if (FAMILY.isEqualTo(type)) {
```

Comment:
this logic has gotten pretty complex, I wonder if it's time to use a strategy pattern. I could be done with the enum strategy approach here: https://www.baeldung.com/a-guide-to-java-enums#2-strategy-pattern

Reply:
Decided to go for a standard strategy pattern, did not want to inject spring beans into enum